NEURON 8.1 breaks LFPy@master

**Describe the bug**
NEURON-8.1 released a few days ago breaks the network simulation codes.

**To Reproduce**
```
% py.test -v LFPy/test/test_network.py::testNetwork::test_Network_00
============================================================================================================ test session starts =============================================================================================================
platform darwin -- Python 3.10.2, pytest-7.1.0, pluggy-1.0.0 -- /Users/espenhagen/miniforge3/envs/lfpydev/bin/python
cachedir: .pytest_cache
rootdir: /Users/espenhagen/Repositories/LFPy
plugins: anyio-3.5.0
collected 1 item                                                                                                                                                                                                                             

LFPy/test/test_network.py::testNetwork::test_Network_00 FAILED                                                                                                                                                                         [100%]

================================================================================================================== FAILURES ==================================================================================================================
________________________________________________________________________________________________________ testNetwork.test_Network_00 _________________________________________________________________________________________________________
TypeError: float() argument must be a string or a real number, not 'hoc.HocObject'

The above exception was the direct cause of the following exception:

self = <LFPy.test.test_network.testNetwork testMethod=test_Network_00>

    def test_Network_00(self):
        cellParameters = dict(
            morphology=os.path.join(
                LFPy.__path__[0],
                'test',
                'ball_and_sticks_w_lists.hoc'),
            templatefile=os.path.join(
                LFPy.__path__[0],
                'test',
                'ball_and_stick_template.hoc'),
            templatename='ball_and_stick_template',
            templateargs=None,
            passive=False,
            dt=2**-3,
            tstop=100,
            delete_sections=False,
        )
    
        populationParameters = dict(
            CWD=None,
            CELLPATH=None,
            Cell=LFPy.NetworkCell,
            cell_args=cellParameters,
            pop_args=dict(
                radius=100,
                loc=0.,
                scale=20.),
            rotation_args=dict(x=0, y=0),
            POP_SIZE=4,
            name='test',
        )
        networkParameters = dict(
            dt=2**-3,
            tstart=0.,
            tstop=100.,
            v_init=-65.,
            celsius=6.3,
            OUTPUTPATH='tmp_testNetworkPopulation'
        )
        # set up
        network = LFPy.Network(**networkParameters)
        network.create_population(**populationParameters)
        connectivity = network.get_connectivity_rand(
            pre='test', post='test', connprob=0.5)
    
        # test set up
        for population in network.populations.values():
            self.assertTrue(len(population.cells) == population.POP_SIZE)
            for cell, soma_pos, gid in zip(
                    population.cells, population.soma_pos, population.gids):
                self.assertTrue(isinstance(cell, LFPy.NetworkCell))
                self.assertTrue((cell.somapos[0] == soma_pos['x']) &
                                (cell.somapos[1] == soma_pos['y']) &
                                (cell.somapos[2] == soma_pos['z']))
                self.assertEqual(cell.gid, gid)
                self.assertTrue(
                    np.sqrt(
                        soma_pos['x']**2 +
                        soma_pos['y']**2) <= 100.)
            np.testing.assert_equal(population.gids, np.arange(4))
    
        np.testing.assert_equal(
            connectivity.shape,
            (population.POP_SIZE,
             population.POP_SIZE))
        np.testing.assert_equal(
            connectivity.diagonal(), np.zeros(
                population.POP_SIZE))
    
        # connect and run sim
        network.connect(pre='test', post='test', connectivity=connectivity)
>       network.simulate()

LFPy/test/test_network.py:164: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <LFPy.network.Network object at 0x127ab59f0>, probes = None, rec_imem = False, rec_vmem = False, rec_ipas = False, rec_icap = False, rec_isyn = False, rec_vmemsyn = False, rec_istim = False, rec_pop_contributions = False
rec_variables = [], variable_dt = False, atol = 0.001, to_memory = True, to_file = False, file_name = 'OUTPUT.h5', kwargs = {}, cvode = CVode[1], dt = 0.125, name = 'test', cell = <LFPy.network.NetworkCell object at 0x127ab5cc0>
population = <LFPy.network.NetworkPopulation object at 0x127ab5a20>, i = 0

    def simulate(self, probes=None,
                 rec_imem=False, rec_vmem=False,
                 rec_ipas=False, rec_icap=False,
                 rec_isyn=False, rec_vmemsyn=False, rec_istim=False,
                 rec_pop_contributions=False,
                 rec_variables=[], variable_dt=False, atol=0.001,
                 to_memory=True, to_file=False,
                 file_name='OUTPUT.h5',
                 **kwargs):
        """
        This is the main function running the simulation of the network model.
    
        Parameters
        ----------
        probes: list of :obj:, optional
            None or list of LFPykit.RecExtElectrode like object instances that
            each have a public method `get_transformation_matrix` returning
            a matrix that linearly maps each compartments' transmembrane
            current to corresponding measurement as
    
            .. math:: \\mathbf{P} = \\mathbf{M} \\mathbf{I}
    
        rec_imem: bool
            If true, segment membrane currents will be recorded
            If no electrode argument is given, it is necessary to
            set rec_imem=True in order to calculate LFP later on.
            Units of (nA).
        rec_vmem: bool
            record segment membrane voltages (mV)
        rec_ipas: bool
            record passive segment membrane currents (nA)
        rec_icap: bool
            record capacitive segment membrane currents (nA)
        rec_isyn: bool
            record synaptic currents of from Synapse class (nA)
        rec_vmemsyn: bool
            record membrane voltage of segments with Synapse (mV)
        rec_istim: bool
            record currents of StimIntraElectrode (nA)
        rec_pop_contributions: bool
            If True, compute and return single-population contributions to
            the extracellular potential during simulation time
        rec_variables: list of str
            variables to record, i.e arg=['cai', ]
        variable_dt: boolean
            use variable timestep in NEURON. Can not be combimed with `to_file`
        atol: float
            absolute tolerance used with NEURON variable timestep
        to_memory: bool
            Simulate to memory. Only valid with `probes=[<probe>, ...]`, which
            store measurements to -> <probe>.data
        to_file: bool
            only valid with `probes=[<probe>, ...]`, saves measurement in
            hdf5 file format.
        file_name: str
            If to_file is True, file which measurements will be
            written to. The file format is HDF5, default is "OUTPUT.h5", put
            in folder Network.OUTPUTPATH
        **kwargs:  keyword argument dict values passed along to function
                    `__run_simulation_with_probes()`, containing some or all of
                    the boolean flags: `use_ipas`, `use_icap`, `use_isyn`
                    (defaulting to `False`).
    
        Returns
        -------
        events
            Dictionary with keys `times` and `gids`, where values are
            ndarrays with detected spikes and global neuron identifiers
    
        Raises
        ------
        Exception
            if `CVode().use_fast_imem()` method not found
        AssertionError
            if rec_pop_contributions==True and probes==None
        """
        # set up integrator, use the CVode().fast_imem method by default
        # as it doesn't hurt sim speeds much if at all.
        cvode = neuron.h.CVode()
        try:
            cvode.use_fast_imem(1)
        except AttributeError:
            raise Exception('neuron.h.CVode().use_fast_imem() not found. '
                            'Please update NEURON to v.7.4 or newer')
    
        # test some of the inputs
        if probes is None:
            assert rec_pop_contributions is False, \
                'rec_pop_contributions can not be True when probes is None'
    
        if not variable_dt:
            dt = self.dt
        else:
            dt = None
    
        for name in self.population_names:
            for cell in self.populations[name].cells:
                cell._set_soma_volt_recorder(dt)
                if rec_imem:
                    cell._set_imem_recorders(dt)
                if rec_vmem:
                    cell._set_voltage_recorders(dt)
                if rec_ipas:
                    cell._set_ipas_recorders(dt)
                if rec_icap:
                    cell._set_icap_recorders(dt)
                if len(rec_variables) > 0:
                    cell._set_variable_recorders(rec_variables)
    
        # run fadvance until t >= tstop, and calculate LFP if asked for
        if probes is None and not rec_pop_contributions and not to_file:
            if not rec_imem:
                if self.verbose:
                    print("rec_imem==False, not recording membrane currents!")
            self.__run_simulation(cvode, variable_dt, atol)
        else:
            self.__run_simulation_with_probes(
                cvode=cvode,
                probes=probes,
                variable_dt=variable_dt,
                atol=atol,
                to_memory=to_memory,
                to_file=to_file,
                file_name='tmp_output_RANK_{:03d}.h5',
                rec_pop_contributions=rec_pop_contributions,
                **kwargs)
    
        for name in self.population_names:
            for cell in self.populations[name].cells:
                # somatic trace
                cell.somav = np.array(cell.somav)
                if rec_imem:
                    cell._calc_imem()
                if rec_ipas:
                    cell._calc_ipas()
                if rec_icap:
                    cell._calc_icap()
                if rec_vmem:
                    cell._collect_vmem()
                if rec_isyn:
                    cell._collect_isyn()
                if rec_vmemsyn:
                    cell._collect_vsyn()
                if rec_istim:
                    cell._collect_istim()
                if len(rec_variables) > 0:
                    cell._collect_rec_variables(rec_variables)
                if hasattr(cell, '_hoc_netstimlist'):
                    del cell._hoc_netstimlist
    
        # Collect spike trains across all RANKs to RANK 0
        for name in self.population_names:
            population = self.populations[name]
            for i in range(len(population.spike_vectors)):
                population.spike_vectors[i] = \
>                   np.array(population.spike_vectors[i])
E               ValueError: setting an array element with a sequence.

LFPy/network.py:1216: ValueError
```

--- a/LFPy/cell.py
+++ b/LFPy/cell.py
@@ -1905,11 +1905,11 @@ class Cell(object):
         Exception
             In case keyword argument ``parent`` is invalid
         """
-        err_mssg = f'keyword argument parent={parent} must be a string'
+        err_mssg = f'keyword argument {parent} must be a string'
         if type(parent) is not str:
             raise ValueError(err_mssg)
         if parent not in self.allsecnames:
-            err_mssg = f'keyword argument parent={parent} not in Cell.allsecnames'
+            err_mssg = f'keyword argument {parent} not in Cell.allsecnames'
             raise ValueError(err_mssg)
 
         seclist = [parent]

--- a/LFPy/cell.py
+++ b/LFPy/cell.py
@@ -1888,18 +1888,30 @@ class Cell(object):
     def get_idx_parent_children(self, parent="soma[0]"):
         """
         Get all idx of segments of parent and children sections, i.e. segment
-        idx of sections connected to parent-argument, and also of the parent
+        idx of sections connected to parent-segment, and also of the parent
         segments
 
         Parameters
         ----------
         parent: str
-            name-pattern matching a sectionname. Defaults to "soma[0]"
+            name-pattern matching a valid section name. Defaults to "soma[0]"
 
         Returns
         -------
         ndarray, dtype=int
+
+        Raises
+        ------
+        Exception
+            In case keyword argument ``parent`` is invalid
         """
+        err_mssg = f'keyword argument parent={parent} must be a string'
+        if type(parent) is not str:
+            raise ValueError(err_mssg)
+        if parent not in self.allsecnames:
+            err_mssg = f'keyword argument parent={parent} not in Cell.allsecnames'
+            raise ValueError(err_mssg)
+
         seclist = [parent]
         for sec in self.allseclist:
             if sec.name() == parent:

--- a/LFPy/network.py
+++ b/LFPy/network.py
@@ -625,7 +625,8 @@ class Network(object):
 
             # record spike events
             population._hoc_spike_vectors.append(neuron.h.Vector())
-            cell._hoc_sd_netconlist[-1].record(population._hoc_spike_vectors[-1])
+            cell._hoc_sd_netconlist[-1].record(
+                population._hoc_spike_vectors[-1])
 
         # add population object to dictionary of populations
         self.populations[name] = population

--- a/LFPy/network.py
+++ b/LFPy/network.py
@@ -361,7 +361,7 @@ class NetworkPopulation(object):
         COMM.Barrier()
 
         # container of Vector objects used to record times of action potentials
-        self.spike_vectors = []
+        self._hoc_spike_vectors = []
 
         # set up population of cells on this RANK
         self.gids = [
@@ -624,8 +624,8 @@ class Network(object):
             self.pc.cell(gid, cell._hoc_sd_netconlist[-1])
 
             # record spike events
-            population.spike_vectors.append(neuron.h.Vector())
-            cell._hoc_sd_netconlist[-1].record(population.spike_vectors[-1])
+            population._hoc_spike_vectors.append(neuron.h.Vector())
+            cell._hoc_sd_netconlist[-1].record(population._hoc_spike_vectors[-1])
 
         # add population object to dictionary of populations
         self.populations[name] = population
@@ -1211,9 +1211,10 @@ class Network(object):
         # Collect spike trains across all RANKs to RANK 0
         for name in self.population_names:
             population = self.populations[name]
-            for i in range(len(population.spike_vectors)):
-                population.spike_vectors[i] = \
-                    np.array(population.spike_vectors[i])
+            population.spike_vectors = []
+            for i in range(len(population._hoc_spike_vectors)):
+                population.spike_vectors += \
+                    [population._hoc_spike_vectors[i].as_numpy()]
         if RANK == 0:
             times = []
             gids = []

--- a/LFPy/test/test_cell.py
+++ b/LFPy/test/test_cell.py
@@ -597,6 +597,17 @@ class testCell(unittest.TestCase):
                 section=[
                     'soma[0]', 'dend[0]']))
 
+    def test_cell_get_idx_parent_children_01(self):
+        cell = LFPy.Cell(morphology=os.path.join(LFPy.__path__[0], 'test',
+                                                 'ball_and_sticks.hoc'))
+        for bad_value in [123, 'no_section']:
+            try:
+                cell.get_idx_parent_children(parent=bad_value)
+            except ValueError:
+                pass
+
+
+
     def test_cell_get_idx_name_00(self):
         cell = LFPy.Cell(morphology=os.path.join(LFPy.__path__[0], 'test',
                                                  'ball_and_sticks.hoc'))

--- a/LFPy/test/test_recextelectrode.py
+++ b/LFPy/test/test_recextelectrode.py
@@ -300,25 +300,25 @@ class testRecExtElectrode(unittest.TestCase):
         stick.z[true_bad_comp, 0] = 1000
         bad_comp, reason = MEA._return_comp_outside_slice()
         np.testing.assert_equal(reason, "zstart above")
-        np.testing.assert_equal(true_bad_comp, bad_comp)
+        np.testing.assert_equal(true_bad_comp, stick.get_idx()[bad_comp])
         stick.z[true_bad_comp, 0] = 100
 
         stick.z[true_bad_comp, 0] = -1000
         bad_comp, reason = MEA._return_comp_outside_slice()
         np.testing.assert_equal(reason, "zstart below")
-        np.testing.assert_equal(true_bad_comp, bad_comp)
+        np.testing.assert_equal(true_bad_comp, stick.get_idx()[bad_comp])
         stick.z[true_bad_comp, 0] = 100
 
         stick.z[true_bad_comp, 1] = 1000
         bad_comp, reason = MEA._return_comp_outside_slice()
         np.testing.assert_equal(reason, "zend above")
-        np.testing.assert_equal(true_bad_comp, bad_comp)
+        np.testing.assert_equal(true_bad_comp, stick.get_idx()[bad_comp])
         stick.z[true_bad_comp, 1] = 100
 
         stick.z[true_bad_comp, 1] = -1000
         bad_comp, reason = MEA._return_comp_outside_slice()
         np.testing.assert_equal(reason, "zend below")
-        np.testing.assert_equal(true_bad_comp, bad_comp)
+        np.testing.assert_equal(true_bad_comp, stick.get_idx()[bad_comp])
         stick.z[true_bad_comp, 1] = 100
 
     def test_position_shifted_slice(self):

